packages/grid/_modules_/grid/components/panel/filterPanel/FilterForm.tsx

```diff
@@ -202,9 +202,10 @@ export function FilterForm(props: FilterFormProps) {
         >
           {currentColumn?.filterOperators?.map((operator) => (
             <option key={operator.value} value={operator.value}>
-              {apiRef!.current.getLocaleText(
-                `filterOperator${capitalize(operator.value)}` as TranslationKeys,
-              )}
+              {operator.label ||
+                apiRef!.current.getLocaleText(
+                  `filterOperator${capitalize(operator.value)}` as TranslationKeys,
```

Comment:
that is weird!
Usually, you want to translate labels not value. This bit needs a learning curve, you need to know that the key is prefixed and uppercased... Why not simply translate the label and leave the value alone? Also should we need to throw if a translation key is missing? Imagine I don't want any translations, just english? I shouldn't have to pass local text, but with this approach it would throw

Reply:
I don't think there is much of a learning curve, the translation keys are easily accessible and can be checked. The reason why I removed the label initially is that it isn't really needed, the key can be build from the value without the need of having another prop. 

For the second concern - if you want to use English you can, there is no need to pass in the `localText` prop. We only throw if we are using a nonexistent translation key internally in the grid (the throw is for us to see if something broke).

Reply:
Well if I create a new operator with value 'between' and I want to translate it, I would have to pass the key `filterOperatorBetween` right?

Reply:
Or I would have to pass the translation directly within the label. Then I would have to manage localization in 2 places, in the prop local-text and in the operators which to me, it does not seem convenient.
If you translate labels, then I would pass my key in label and know what to translate in localtext.

Reply:
> if I create a new operator with value 'between' and I want to translate it

@dtassone I think that we are back to https://github.com/mui-org/material-ui-x/pull/1024#issuecomment-777452741. I think that we should encourage developers to use the `label` prop and translate it when needed. `localeText` is about translating existing part of the UIs. It shouldn't be used for translating customizations.

Reply:
I would pass a key in label and override the translation with the local text...
Otherwise I would have to 

```tsx
const filterOperators = [{
  label: i18n.translate('Between'),
  value: 'between',
}];

<DataGrid columns={[{ id: 'foo', filterOperators }]} localtext = {{filterOperatorContains: 'has'...}}

```

I would have expected 

```tsx
const filterOperators = [{
  label: 'OperatorLabelBetween',
  value: 'between',
}];

<DataGrid columns={[{ id: 'foo', filterOperators }]} localtext = {{OperatorLabelBetween: 'between', OperatorLabelContains: 'has'...}}
```

Reply:
If I was working on a product, I would personally do 1. without any hesitation. But I can imagine it can depend on the developers. @DanailH do you have a preference?